### PR TITLE
Fixes to implement missing functionality

### PR DESCRIPTION
**Pull request by fedorov**
the commit comments should be sufficiently detailed to explain each change

### CODE REVIEW

**Comment by fedorov** [comment]
Remaining item: add test data for --useROIBSpline mode, and add a test to BRAINSFit TestSuite (pending BRAINS Midas write approval, I emailed Dave today).

File: BRAINSCommonLib/BRAINSFitHelperTemplate.hxx [hunk 1]
```diff
@@ -625,6 +632,7 @@ BRAINSFitHelperTemplate<FixedImageType, MovingImageType>::FitCommonCode(
   catch( itk::ExceptionObject& err )
     {
     // pass exception to caller
+    std::cerr << "Exception caught: " << err << std::endl;
```
Comment (dmwelch) [inline BRAINSCommonLib/BRAINSFitHelperTemplate.hxx]: Raise itkGenericExceptionMacro here

File: BRAINSCommonLib/BRAINSFitHelperTemplate.hxx [hunk 2]
```diff
@@ -1173,325 +1181,91 @@ BRAINSFitHelperTemplate<FixedImageType, MovingImageType>::Update(void)
...
           {
-          // As recommended in documentation in
-          // itkMattesMutualInformationImageToImageMetric
-          // "UseExplicitPDFDerivatives = False ... This method is well suited
-          // for Transforms with a large number of parameters, such as,
-          // BSplineDeformableTransforms."
-          const bool UseExplicitPDFDerivatives =
-            ( m_UseExplicitPDFDerivativesMode != "ON" || m_UseExplicitPDFDerivativesMode == "AUTO" ) ? false : true;
-          test_MMICostMetric->SetUseExplicitPDFDerivatives(UseExplicitPDFDerivatives);
+            std::cerr << "ERROR: ROIBSpline mode can only be used with ROI(s) specified!" << std::endl;
```
Comment (dmwelch) [inline BRAINSCommonLib/BRAINSFitHelperTemplate.hxx]: Raise itkGenericExceptionMacro instead

File: BRAINSCommonLib/BRAINSFitHelperTemplate.hxx [hunk 3]
```diff
@@ -1173,325 +1169,91 @@ BRAINSFitHelperTemplate<FixedImageType, MovingImageType>::Update(void)
...
-         * std::cout << "Image size: " <<
-         *    m_FixedVolume->GetBufferedRegion().GetSize() << std::endl;
-         * std::cout << "ROI size: " << roiRegion.GetSize() << std::endl;
-         * std::cout << "ROI spacing: " << roiSpacing << std::endl;
-         * std::cout << "ROI index: " << roiRegion.GetIndex() << std::endl;
-         * std::cout << "ROI size in physical space: " <<
-         * roiRegion.GetSize()[0]*roiSpacing[0] << " " <<
-         * roiRegion.GetSize()[1]*roiSpacing[1] << " " <<
-         * roiRegion.GetSize()[2]*roiSpacing[2] << std::endl;
-         */
```
Comment (dmwelch) [inline BRAINSCommonLib/BRAINSFitHelperTemplate.hxx:1463]: std::cout

File: BRAINSCommonLib/BRAINSFitHelperTemplate.hxx [hunk 4]
```diff
@@ -1613,13 +1371,18 @@ BRAINSFitHelperTemplate<FixedImageType, MovingImageType>::Update(void)
               {
```
Comment (dmwelch) [inline BRAINSCommonLib/BRAINSFitHelperTemplate.hxx:1369]: Good documentation!

File: BRAINSFit/BRAINSFit.cxx [hunk 5]
```diff
@@ -430,7 +430,7 @@ int main(int argc, char *argv[])
     }
   else if( maskProcessingMode == "ROI" )
     {
-    if( fixedBinaryVolume == "" || movingBinaryVolume == "")
+    if( fixedBinaryVolume == "" && movingBinaryVolume == "")
```
Comment (dmwelch) [inline BRAINSFit/BRAINSFit.cxx:433]: @hjmjohnson: This looks correct - can you see any reason why this was an "or" statement?  I'm going to verify that we have a test using just one mask, and if not, add one.

File: BRAINSFit/TestSuite/CMakeLists.txt [hunk 6]
```diff
@@ -1694,5 +1694,41 @@ ExternalData_add_test( ${PROJECT_NAME}FetchData NAME ${BRAINSFitEZTestName}
   --debugLevel 50
 )
 
+# Test for ROIBSpline mode
+# TODO: Test xml parsing instead of algorithm (that's done with BRAINSFit tests)
```
Comment (dmwelch) [inline BRAINSFit/TestSuite/CMakeLists.txt]: Remove TODO

**Pull request merged**